In [1]:
print("all good?")

all good?


In [2]:
import os

In [3]:
from dotenv import load_dotenv

In [4]:
load_dotenv()

True

In [5]:
from autogen_agentchat.agents import AssistantAgent
from autogen_ext.models.openai import OpenAIChatCompletionClient
from dotenv import load_dotenv
import os

In [6]:
load_dotenv()

api_key=os.getenv("OPENAI_API_KEY")

In [7]:
model_client=OpenAIChatCompletionClient(model='gpt-4o',api_key=api_key)

In [8]:
agent_1=AssistantAgent(name='my_assistant',model_client=model_client)

In [9]:
result =await agent_1.run(task='Tell me something about you.')

In [10]:
print(result)

messages=[TextMessage(id='26102440-a08a-427f-a017-1d871facf286', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 20, 3, 35, 26, 861420, tzinfo=datetime.timezone.utc), content='Tell me something about you.', type='TextMessage'), TextMessage(id='ec26013f-4164-425c-ace9-780c894fe237', source='my_assistant', models_usage=RequestUsage(prompt_tokens=42, completion_tokens=98), metadata={}, created_at=datetime.datetime(2025, 7, 20, 3, 35, 31, 131901, tzinfo=datetime.timezone.utc), content="I am an AI developed to assist with a wide range of tasks by providing information, answering questions, and performing problem-solving activities. My goal is to help users by leveraging the data I've been trained on, which includes a diverse array of topics up until October 2023. I don't have personal experiences or emotions, but I'm here to help make your life easier by providing accurate and helpful responses to your inquiries. If there's anything specific you'd like a

messages=[TextMessage(id='26102440-a08a-427f-a017-1d871facf286', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 20, 3, 35, 26, 861420, tzinfo=datetime.timezone.utc), content='Tell me something about you.', type='TextMessage'), TextMessage(id='ec26013f-4164-425c-ace9-780c894fe237', source='my_assistant', models_usage=RequestUsage(prompt_tokens=42, completion_tokens=98), metadata={}, created_at=datetime.datetime(2025, 7, 20, 3, 35, 31, 131901, tzinfo=datetime.timezone.utc), content="I am an AI developed to assist with a wide range of tasks by providing information, answering questions, and performing problem-solving activities. My goal is to help users by leveraging the data I've been trained on, which includes a diverse array of topics up until October 2023. I don't have personal experiences or emotions, but I'm here to help make your life easier by providing accurate and helpful responses to your inquiries. If there's anything specific you'd like assistance with, feel free to let me know!", type='TextMessage')] stop_reason=None


In [13]:
from autogen_core.models import UserMessage
from autogen_ext.models.openai import OpenAIChatCompletionClient
load_dotenv()
model_client = OpenAIChatCompletionClient(
    model="gemini-1.5-flash-8b",
    api_key=os.getenv("GEMENI_APIKEY"),
)

response = await model_client.create([UserMessage(content="What is the capital of France?", source="user")])
print(response)
await model_client.close()


finish_reason='stop' content='Paris\n' usage=RequestUsage(prompt_tokens=7, completion_tokens=2) cached=False logprobs=None thought=None
